cmd/anonymizer/app/writer/writer_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package writer
+
+import (
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/model"
+)
+
+var tags = []model.KeyValue{
+	model.Bool("error", true),
+	model.String("http.method", "POST"),
+	model.Bool("foobar", true),
+}
+
+var traceID = model.NewTraceID(1, 2)
+
+var span = &model.Span{
+	TraceID: traceID,
+	SpanID:  model.NewSpanID(1),
+	Process: &model.Process{
+		ServiceName: "serviceName",
+		Tags:        tags,
+	},
+	OperationName: "operationName",
+	Tags:          tags,
+	Logs: []model.Log{
+		{
+			Timestamp: time.Now(),
+			Fields: []model.KeyValue{
+				model.String("logKey", "logValue"),
+			},
+		},
+	},
+	Duration:  time.Second * 5,
+	StartTime: time.Unix(300, 0),
+}
+
+func TestNew(t *testing.T) {
+	nopLogger := zap.NewNop()
+	tempDir := t.TempDir()
+
+	tests := []struct {
+		name   string
+		config Config
+	}{
+		{
+			name: "successfully create writer",
+			config: Config{
+				MaxSpansCount:  10,
+				CapturedFile:   tempDir + "/captured.json",
+				AnonymizedFile: tempDir + "/anonymized.json",
+				MappingFile:    tempDir + "/mapping.json",
+			},
+		},
+		{
+			name: "captured.json doesn't exist",
+			config: Config{
+				CapturedFile:   tempDir + "/nonexistent_directory/captured.json",
+				AnonymizedFile: tempDir + "/anonymized.json",
+				MappingFile:    tempDir + "/mapping.json",
+			},
+		},
+		{
+			name: "anonymized.json doesn't exist",
+			config: Config{
+				CapturedFile:   tempDir + "/captured.json",
+				AnonymizedFile: tempDir + "/nonexistent_directory/anonymized.json",
+				MappingFile:    tempDir + "/mapping.json",
+			},
+		},
+	}
+
+	t.Run(tests[0].name, func(t *testing.T) {
+		_, err := New(tests[0].config, nopLogger)
+		require.NoError(t, err)
+	})
+	t.Run(tests[1].name, func(t *testing.T) {
+		_, err := New(tests[1].config, nopLogger)
+		require.Error(t, err)
+	})
+	t.Run(tests[2].name, func(t *testing.T) {
+		_, err := New(tests[2].config, nopLogger)
+		require.Error(t, err)
+	})
+}
+
+func TestWriter_WriteSpan(t *testing.T) {
+	nopLogger := zap.NewNop()
+	tempDir := t.TempDir()
+
+	config := Config{
+		MaxSpansCount:  101,
+		CapturedFile:   tempDir + "/captured.json",
+		AnonymizedFile: tempDir + "/anonymized.json",
+		MappingFile:    tempDir + "/mapping.json",
+	}
+
+	writer, err := New(config, nopLogger)
+	require.NoError(t, err)
+
+	for i := 0; i <= 99; i++ {
+		err = writer.WriteSpan(span)
+		require.NoError(t, err)
+	}
+}
+
+func TestWriter_WriteSpan_Exits(t *testing.T) {
+	if os.Getenv("BE_SUBPROCESS") == "1" {
```

Comment:
where would this be set?

Reply:
This is set on line 141. Actually the code I was testing here was exiting with `os.Exit(0)` so the tests were not running properly. My first thought was to look for tests dealing with `os.Exit()` in the codebase itself, however there were none. So I referred to an [article](https://sr-taj.medium.com/how-to-test-methods-that-kill-your-program-in-golang-e3b879185b8a) and found this implementation. If this is incorrect let me know I will look for other methods and fix it.